cudamapper/src/overlapper.cpp

```diff
@@ -12,17 +12,31 @@
 #include "index_cpu.hpp"
 
 namespace claragenomics {
+
+    std::vector<Overlap> Overlapper::filter_overlaps(const std::vector<Overlap> &overlaps,size_t  min_residues, size_t  min_overlap_len) {
+        std::vector<Overlap> filtered_overlaps;
+        for(auto overlap: overlaps){
+            if ((overlap.num_residues_ >= min_residues) &&
+                ((overlap.query_end_position_in_read_ - overlap.query_start_position_in_read_) > min_overlap_len)){
+                filtered_overlaps.push_back(overlap);
+            }
+        }
+        return filtered_overlaps;
+    }
+
     void Overlapper::print_paf(const std::vector<Overlap> &overlaps){
+        std::vector<Overlap> filtered_overlaps = filter_overlaps(overlaps);
+
         std::string relative_strand = "+";
-        for(const auto& overlap: overlaps){
+        for(const auto& overlap: filtered_overlaps){
```

Comment:
This is a performance vs clean code discussion, but how about moving the functionality of `filter_overlaps` here and avoiding building a completely new vector?

Reply:
another option is to remove the constnesss of `overlaps` and have `filter_overlaps` do an in-place filtering? IMO the performance impact of this is likely so negligible that we should leave it as is for now.

Reply:
Yes, doing it in-place would be an option. Have a look at https://en.wikipedia.org/wiki/Erase%E2%80%93remove_idiom

Reply:
Or as you suggest in another discussion you can make another function/class/lambda which returns next valid element (i.e. the one which would not be filtered out) on each call. This would actually probably have the best performance and lowest memory consumption

Reply:
@mimaric I suggest a separate PR for this, what do you think?

Reply:
Yes, let's keep it like it is for now